lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -6120,59 +6120,66 @@ void swift::checkOverrideActorIsolation(ValueDecl *value) {
 }
 
 void swift::checkGlobalIsolation(VarDecl *var) {
+  // Skip this if the relevant features aren't supported.
+  if (!var->getLoc() ||
+      !var->getASTContext().LangOpts.hasFeature(Feature::GlobalConcurrency))
+    return;
+
+  // Skip if the decl is global actor-isolated or is unsafely opted-out.
   const auto isolation = getActorIsolation(var);
-  if (var->getLoc() &&
-      var->getASTContext().LangOpts.hasFeature(Feature::GlobalConcurrency) &&
-      !isolation.isGlobalActor() &&
-      (isolation != ActorIsolation::NonisolatedUnsafe)) {
-    auto *classDecl = var->getDeclContext()->getSelfClassDecl();
-    const bool isActorType = classDecl && classDecl->isAnyActor();
-    if (var->isGlobalStorage() && !isActorType) {
-      auto *diagVar = var;
-      if (auto *originalVar = var->getOriginalWrappedProperty()) {
-        diagVar = originalVar;
-      }
+  if (isolation.isGlobalActor() || isolation.isNonisolatedUnsafe())
```

Comment:
minor change here to use the `isNonisolatedUnsafe()` accessor vs what was previously being done.

tangent: if someone cares to enlighten me as to whether these two formulations are actually the same i'd be curious to learn more about exactly why/how that is. the first case seems like a class is being compared to a 'Kind' which i don't entirely understand.

Reply:
I believe the reason the comparison between `ActorIsolation` and `ActorIsolation::Kind` works is because there's an operator `()` on `ActorIsolation` that returns its kind:

```cpp
operator Kind() const { return getKind(); }
```

I believe operator `()` was implicitly called on `ActorIsolation` when used as an operand to `==` or `!=`, so it was really just a comparison between two `ActorIsolation::Kind`s.